### ISSUE TITLE
Install Arduino-CLI when not already present

### ISSUE THREAD

**Issue by PeterWone** [PeterWone/vscode-arduino-cli#5]
On *nix systems the following will install the CLI
``` 
https://arduino.github.io/arduino-cli/latest/installation/#use-the-install-script
```

On Windows systems it is necessary to determine platform bitness and fetch one of

* https://downloads.arduino.cc/arduino-cli/arduino-cli_latest_Windows_64bit.zip
* https://downloads.arduino.cc/arduino-cli/arduino-cli_latest_Windows_32bit.zip

then unpack the file and put the executable somewhere in the filesystem, then set the extension path accordingly.

**Comment by RBM-PITT** [comment]
"put the executable somewhere on the system path" is this a requirement? 

here's the basis for that question: 

Does having the extension setting for the path handle locating the arduino-cli.exe vs relying on having the executable in the system path?

![image](https://user-images.githubusercontent.com/24991434/97083569-e7c06200-15de-11eb-9c35-713088ae6be0.png)

**Comment by PeterWone** [comment]
No, I've re-worded that now. What I meant was after you put the executable in the local file-system you have to tell the extension where it is using the setting you snapped.

If you were to put `arduino-cli.exe` on the system path then you could probably abbreviate the path setting to just the name of the executable for `child_process.exec()` but `child_process.spawn()` doesn't shell, it runs the exe directly so this may not work.

Thus, best to supply a fully qualified path in the path setting _but_ as a courtesy and convenience to the user it should also be on the system path.

**Comment by PeterWone** [comment]
Node reports Windows platforms as win32 irrespective of their actual bitness but `child_process.execSync("echo %processor_architecture%") ` returns "AMD64" for 64 bit Windows.

**Comment by PeterWone** [comment]
@RBM-PITT a note on your path, it should be the path to the executable, not the path to the folder containing the executable

```
/path/to/arduino-cli.exe
c:\path\to\arduino-cli.exe
```

**Comment by PeterWone** [comment]
@RBM-PITT These instructions are now in the wiki. Feedback? 

https://github.com/PeterWone/vscode-arduino-cli/wiki/How-to-manually-install-Arduino-CLI

**Comment by RBM-PITT** [comment]
@PeterWone This is perfect - love it. I should be able to step through it later today. Do we have the ability to contribute to the Wiki? I could help out by adding in detail directly rather than having you do that.

**Comment by PeterWone** [comment]
Um... I dunno? Try!

**Comment by PeterWone** [comment]
I have made you a collaborator. This should allow you to create pages as well as edit the existing material. It probably also permits accidental destruction so take care until we understand the tools better.

As a side note, this issue wasn't intended as an instruction for use, it is a feature for implementation, that's why I didn't originally write it in a user friendly way.

**Comment by RBM-PITT** [comment]
@PeterWone I can access the Wiki to contribute - thanks, I understand the need to watch for deletions.

I think this issue can close.
